crates/storage/src/snapshot.rs

```diff
@@ -112,8 +126,8 @@ impl Snapshot {
     /// Reads from the JMT will fail if the root is missing; this method
     /// special-cases the empty tree case so that reads on an empty tree just
     /// return None.
-    fn get_jmt(&self, key: jmt::KeyHash) -> Result<Option<Vec<u8>>> {
-        let tree = Sha256Jmt::new(&*self.0);
+    fn get_jmt(&self, key: jmt::KeyHash, substore: SubstoreSnapshot) -> Result<Option<Vec<u8>>> {
+        let tree = Sha256Jmt::new(&substore);
```

Comment:
Why is this a method on `Snapshot` taking the `SubstoreSnapshot` that will actually be used, instead of moving the method out to the `SubstoreSnapshot`? Wouldn't it be cleaner to move all of the tree accesses out to the `SubstoreSnapshot`?